src/components/FilterBar/FilterDisplay/Inputs/RelativeDateInput.react.js

```diff
@@ -4,49 +4,81 @@ export class RelativeDateInput extends React.Component {
     super(props);
     this.state = { value: this.props.value || { from: null, to: null, value: null } };
+    this.setDisplayDates(this.props.value['value']);
   }
 
-  onRelativeChange(event) {
-    var optionElement = event.target.childNodes[event.target.selectedIndex];
-    var from = moment(parseInt(optionElement.getAttribute('data-from')));
-    var to = moment(parseInt(optionElement.getAttribute('data-to')));
-    var newValue = { value: optionElement.value, from: from.format(this.props.dateFormat), to: to.format(this.props.dateFormat) };
+  // If relative option selected, set dates for the datepickers to display
+  setDisplayDates(relativeDateSelection) {
+    if(relativeDateSelection === undefined || relativeDateSelection == '') {
+      return;
+    }
+
+    var selected            = this.relativeOptions()[relativeDateSelection];
+    this.state.displayFrom  = selected.from.format(this.props.dateFormat);
+    this.state.displayTo    = selected.to.format(this.props.dateFormat);
+  }
 
-    this.setState({ value: newValue });
+  onRelativeChange(event) {
+    var selectedOption = $(event.target.childNodes[event.target.selectedIndex]);
+    var newValue = { value: selectedOption.val() };
+    this.state = { value: newValue }
     this.updateFilter(newValue);
   }
 
+  onDatePickerChange(event) {
+    var newValue = {
+      from: this.state.value.from || this.state.displayFrom,
+      to: this.state.value.to || this.state.displayTo
+    };
+
+    if(event.type === "dp") {
+      newValue[event.target.querySelector("input").getAttribute("placeholder")] = event.target.querySelector("input").value;
+    } else if (event.type === "input") {
+      newValue[event.target.getAttribute("placeholder")] = event.target.value;
+    }
+
+    this.setState({value: newValue});
+  }
+
   updateFilter(newValue) {
     this.context.filterBarActor.updateFilter(this.props.filterUid, "value", newValue);
   }
 
-  // TODO: Update relative dates based on relative selection (if in query params) on page load, rather than applying stored dates directly
   relativeOptions() {
     var lastWeek = moment().subtract(1, 'week');
-    const optionsList = [
-      { label: 'Select Period' , from: null , to: null },
-      { label: 'Today' , from: moment() , to: moment() },
-      { label: 'Last week' , from: lastWeek.clone().startOf('isoWeek'), to: lastWeek.clone().endOf('isoWeek') },
-      { label: 'This week' , from: moment().startOf('isoWeek'), to: moment().endOf('isoWeek') },
-    ]
+    const optionsList = {
+      'Select Period':  { value: '',  from: null , to: null },
+      'Today':          { from: moment() , to: moment() },
+      'Last Week':      { from: lastWeek.clone().startOf('isoWeek'), to: lastWeek.clone().endOf('isoWeek') },
+      'This week':      { from: moment().startOf('isoWeek'), to: moment().endOf('isoWeek') },
+    }
     return optionsList
   }
 
+  relativeOption(optionKey) {
+    var optionItem = this.relativeOptions()[optionKey];
```

Comment:
This causes us to rebuild the options for every key. Should we instead cache them in the state in the constructor?